File: TDDBC_VendorMachine/ShinsukeAbe/products/VendingMachine/src/java/com/codersation/abe/vendingmachine/managestock/Juice.java
package com.codersation.abe.vendingmachine.managestock;

/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 *
 * @author mao
 */
public class Juice {
    private String name;
    private Integer price;
    
    public Juice(String name, Integer price) {
        this.name = name;
        this.price = price;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getPrice() {
        return price;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }
    
    @Override
    public boolean equals(Object that) {
        if(that instanceof Juice) {
            Juice thatJuice = (Juice)that;
            if(this.name.equals(thatJuice.getName())
                    && this.price.equals(thatJuice.getPrice())) {
                return true;
            } else {
                return false;
            }
        } else {
            return false;
        }
        
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 67 * hash + (this.name != null ? this.name.hashCode() : 0);
        hash = 67 * hash + (this.price != null ? this.price.hashCode() : 0);
        return hash;
    }
    
}

File: TDDBC_VendorMachine/HappyLuckyAkira/VendingMachine/Product/Juice.cpp
/*
 * Juice.cpp
 *
 *  Created on: 2012/10/07
 *      Author: Akira
 */

#include "Juice.h"

//Juiceのコンストラクタ
Juice::Juice(const std::string& name, int price)
:name_(name),price_(price)
{
}

bool
Juice::operator ==(const Juice& rjuice) const
{
  if (name_ == rjuice.getName() && price_ == rjuice.getPrice()) return true;
  return false;
}

bool
Juice::operator !=(const Juice& rjuice) const
{
  if (name_ != rjuice.getName() || price_ != rjuice.getPrice()) return true;
  return false;
}

//C++11なら、deleteしておくところ
Juice::Juice()
{

}

Juice::~Juice()
{
}

